usaspending_api/reporting/v2/views/agencies/agency_code/overview.py

```diff
@@ -0,0 +1,127 @@
+from django.db.models import Subquery, OuterRef, DecimalField, Func, F, Q, IntegerField
+from rest_framework.response import Response
+from usaspending_api.agency.v2.views.agency_base import AgencyBase
+from django.utils.functional import cached_property
+
+from usaspending_api.common.data_classes import Pagination
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.common.validator import customize_pagination_with_sort_columns, TinyShield
+from usaspending_api.reporting.models import ReportingAgencyOverview, ReportingAgencyTas, ReportingAgencyMissingTas
+from usaspending_api.submissions.models import SubmissionAttributes
+
+
+class AgencyOverview(AgencyBase):
+    """Returns an overview of the specified agency's submission data"""
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/reporting/agencies/agency_code/overview.md"
+
+    def get(self, request, toptier_code):
+        results = self.get_agency_overview()
+        page_metadata = get_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        results = results[self.pagination.lower_limit : self.pagination.upper_limit]
+        return Response(
+            {
+                "page_metadata": page_metadata,
+                "results": results[: self.pagination.limit],
```

Comment:
I would think line 21 is enough to obtain the correct subset of results to return

Reply:
That is true